File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        b_list=[]
        max_candies = max(candies)
        i=0
        while i<len(candies):
            if not candies[i]+extraCandies >= max_candies:
                b_list.append(False)
            else:
                b_list.append(True)
            i+=1
        return b_list
    # OR
        # max_candies = max(candies)  # Find the maximum number of candies initially
        # result = [candy + extraCandies >= max_candies for candy in candies]
        # return result

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> b_list = new ArrayList<>();
        int maxCandies = getMax(candies);
        for(int i =0; i<candies.length; i++){
            if(candies[i]+extraCandies>=maxCandies)
                b_list.add(true);
            else
                b_list.add(false);
        }
        return b_list;
        
    }
    private int getMax(int[] candies){
        int maxCandies = 0;
        for (int i=0; i<candies.length; i++){
            if(candies[i]>maxCandies)
                maxCandies=candies[i];
        }
        return maxCandies;
    }
}